lib/active_record_doctor/detectors/missing_unique_indexes.rb

```diff
@@ -116,6 +123,26 @@ def ignore_columns
           column.gsub(" ", "")
         end
       end
+
+      def expression_indexes_unsupported?
+        (ActiveRecord::VERSION::STRING < "5.0") ||
+          # Active Record < 6 is unable to correctly parse expression indexes for MySQL.
+          (mysql? && ActiveRecord::VERSION::STRING < "6.0")
+      end
```

Comment:
I suggest extracting this to `ActiveRecordDoctor.expression_indexes_unsupported?` and using it in this detector and the test suite.

Reply:
I extracted a separate module for this kind of utils. It feels not correct to me to pollute public `ActiveRecordDoctor` namespace with such helper functions. Let me know what you think.